src/Illuminate/Auth/Access/Gate.php

```diff
@@ -275,7 +275,7 @@ protected function resolvePolicyCallback($user, $ability, array $arguments)
                 }
             }
 
-            if (! method_exists($instance, $ability)) {
+            if (! is_callable([$instance, $ability])) {
```

Comment:
I can add that if you like. I don't see myself ever overloading `before()`, but then never say never.